roq-management/src/main/java/org/roqmessaging/management/GlobalConfigurationManager.java

```diff
@@ -178,12 +205,54 @@ public void run() {
 				logger.debug("Receiving request..." + content);
 				
 				//check if the string contains a "," if not that means that it is a BSON encoded request
-				if(content.contains(",")){
+				if(content.contains(",")) {
 					processStandardRequest(content);
 				}else{
 					processBSONRequest(encoded);
 				}
 			}
+			// If connection with zookeeper has been lost
+			if (!hasLead) {
```

Comment:
Can you please refactor the code in this ```if``` statement? It probably works, but some of these operations should have their own functions, which would allow you to also simplify the ```init()``` function's code.